### PR TITLE
pod: add WaitForAllPodsInNamespacesHealthy

### PR DESCRIPTION
**Pull request by edcdavid**
Refactoring functions previously defined in eco-gotests RAN subfolder:

WaitForAllPodsInNamespacesHealthy https://github.com/openshift-kni/eco-gotests/blob/050a4e5ee720e9a776a90791c8354ac9c5aaf43e/tests/cnf/ran/talm/internal/helper/cluster.go#L146. New function name is: WaitForAllPodsInNamespacesHealthy

### CODE REVIEW

**Review by kononovn — DISMISSED** [review]
/lgtm

File: pkg/pod/list.go [hunk 1]
```diff
@@ -182,3 +207,85 @@ func WaitForAllPodsInNamespaceRunning(
 
 	return true, nil
 }
+
+// WaitForAllPodsInNamespacesHealthy waits until all pods in a list of namespaces that match options
```
Comment (achuzhoy) [inline pkg/pod/list.go:211]:
```suggestion
// WaitForAllPodsInNamespacesHealthy waits until:
- all pods in a list of namespaces that match options are in healthy state.
-  a pod in a healthy state is in running phase and optionally in ready condition.
```

File: pkg/pod/list.go [hunk 2]
```diff
@@ -182,3 +207,85 @@ func WaitForAllPodsInNamespaceRunning(
...
+// are in healthy state.
+// An pod in a healthy state is in running phase and optionally in ready condition.
+//
+// nsNames lists the list of namespaces to monitor. Monitors all namespaces when empty.
+// timeout is the duration to wait for the pods to be healthy
+// includeSucceeded when true, implies that pods in succeeded phase are running.
```
Comment (achuzhoy) [inline pkg/pod/list.go:217]:
```suggestion
// includeSucceeded when true, checks that pods in succeeded phase are running.
```

File: pkg/pod/list.go [hunk 3]
```diff
@@ -182,3 +207,85 @@ func WaitForAllPodsInNamespaceRunning(
...
+// checkReadiness when true, to also checks that the podConditions are ready.
```
Comment (achuzhoy) [inline pkg/pod/list.go:218]:
```suggestion
// checkReadiness when true, checks that the podConditions are ready.
```

File: pkg/pod/list.go [hunk 4]
```diff
@@ -182,3 +207,85 @@ func WaitForAllPodsInNamespaceRunning(
...
+// ignoreFailedPods when true, to Ignore failed pods with restart policy set to never.
```
Comment (achuzhoy) [inline pkg/pod/list.go:219]:
```suggestion
// ignoreFailedPods when true, ignores failed pods with restart policy set to never.
```

File: pkg/pod/list.go [hunk 5]
```diff
@@ -182,3 +207,85 @@ func WaitForAllPodsInNamespaceRunning(
...
+	passedOptions := metav1.ListOptions{}
+
+	if apiClient == nil {
+		glog.V(100).Infof("The apiClient is empty")
+
+		return fmt.Errorf("podList 'apiClient' cannot be empty")
+	}
+
+	if len(options) > 1 {
+		glog.V(100).Infof("'options' parameter must be empty or single-valued")
```
Comment (achuzhoy) [inline pkg/pod/list.go:242]: shouldn;t the options be option (single) if it only supports a single option

File: pkg/pod/list.go (hunk 1 above)
Comment (edcdavid) [reply pkg/pod/list.go]: Fixed

File: pkg/pod/list.go [hunk 6]
```diff
@@ -182,3 +207,85 @@ func WaitForAllPodsInNamespaceRunning(
 
 	return true, nil
 }
+
+// WaitForAllPodsInNamespacesHealthy waits until all pods in a list of namespaces that match options
+// are in healthy state.
+// An pod in a healthy state is in running phase and optionally in ready condition.
+//
+// nsNames lists the list of namespaces to monitor. Monitors all namespaces when empty.
+// timeout is the duration to wait for the pods to be healthy
```
[the remark this answers is not in the document]
Comment (edcdavid) [reply pkg/pod/list.go]: Fixed

File: pkg/pod/list.go (hunk 2 above)
Comment (edcdavid) [reply pkg/pod/list.go]: Fixed. Reworded to clarify the intent. includeSucceeded considers succeeded pods as healthy

File: pkg/pod/list.go (hunk 3 above)
Comment (edcdavid) [reply pkg/pod/list.go]: Fixed. Aligned parameter name/meaning

File: pkg/pod/list.go (hunk 4 above)
Comment (edcdavid) [reply pkg/pod/list.go]: Fixed. Aligned parameter name/meaning

File: pkg/pod/list.go (hunk 5 above)
Comment (edcdavid) [reply pkg/pod/list.go:242]: I followed the pattern in other fucntions such as ListInAllNamespaces, etc

File: pkg/pod/list.go [hunk 7]
```diff
@@ -182,3 +207,85 @@ func WaitForAllPodsInNamespaceRunning(
 
 	return true, nil
 }
+
+// WaitForAllPodsInNamespacesHealthy waits until:
+// - all pods in a list of namespaces that match options are in healthy state.
+// - a pod in a healthy state is in running phase and optionally in ready condition.
+//
+// nsNames lists the list of namespaces to monitor. Monitors all namespaces when empty.
```
Comment (achuzhoy) [inline pkg/pod/list.go:215]:
```suggestion
// nsNames passes the list of namespaces to monitor. Monitors all namespaces when empty.
```

**Review by achuzhoy — DISMISSED** [review]

**Comment by achuzhoy** [comment]
minor text suggestion.
/lgtm

File: pkg/pod/list.go (hunk 7 above)
Comment (edcdavid) [reply pkg/pod/list.go]: Fixed.

**Review by achuzhoy — DISMISSED** [review]

**Review by achuzhoy — APPROVED** [review]

**Pull request merged**